cli/container_test.go

```diff
@@ -329,3 +329,97 @@ func Test_parseDeviceMappings(t *testing.T) {
 		})
 	}
 }
+
+func Test_parseSysctls(t *testing.T) {
+	type result struct {
+		sysctls map[string]string
+		err     error
+	}
+	type TestCases struct {
+		input  []string
+		expect result
+	}
+
+	testCases := []TestCases{
+		{
+			input: []string{"a=b"},
+			expect: result{
+				sysctls: map[string]string{"a": "b"},
+				err:     nil,
+			},
+		},
+		{
+			input: []string{"ab"},
+			expect: result{
+				sysctls: nil,
+				err:     fmt.Errorf("invalid sysctl: %s: sysctl must be in format of key=value", "ab"),
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		sysctl, err := parseSysctls(testCase.input)
+		assert.Equal(t, testCase.expect.sysctls, sysctl)
+		assert.Equal(t, testCase.expect.err, err)
+	}
+}
+
+func Test_parseNetwork(t *testing.T) {
+	type net struct {
+		name string
+		ip   string
+	}
+	type result struct {
+		network net
+		err     error
+	}
+	type TestCases struct {
+		input  string
+		expect result
+	}
+
+	testCases := []TestCases{
+		{
+			input: "",
+			expect: result{
+				err:     fmt.Errorf("invalid network: cannot be empty"),
+				network: net{name: "", ip: ""},
+			},
+		},
+		{
+			input: "121.0.0.1",
+			expect: result{
+				err:     nil,
+				network: net{name: "", ip: "121.0.0.1"},
+			},
+		},
+		{
+			input: "myHost",
+			expect: result{
+				err:     nil,
+				network: net{name: "myHost", ip: ""},
+			},
+		},
+		{
+			input: "myHost:121.0.0.1",
+			expect: result{
+				err:     nil,
+				network: net{name: "myHost", ip: "121.0.0.1"},
+			},
+		},
+		{
+			input: "myHost:myHost",
+			expect: result{
+				err:     fmt.Errorf("invalid network ip: %s", "myHost"),
+				network: net{name: "", ip: ""},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		name, ip, error := parseNetwork(testCase.input)
+		assert.Equal(t, testCase.expect.err, error)
+		assert.Equal(t, testCase.expect.network.name, name)
+		assert.Equal(t, testCase.expect.network.ip, ip)
+	}
```

Comment:
@allencloud However, we need to know whether or not the type of the error is the exact one we are expecting. In this case, I don't think the unit test framework generated by vscode is good enough.